lib/workload/stateful/database/component.ts

```diff
@@ -16,59 +18,120 @@ type MonitoringProps = {
    */
   readonly enablePerformanceInsights?: boolean;
   /**
-   * performance insights retention period
+   * performance insights retention period.
    */
   readonly performanceInsightsRetention?: rds.PerformanceInsightRetention;
   /**
-   * Enable enhanced monitoring by specifying the interval
+   * Enable enhanced monitoring by specifying the interval.
    */
   readonly enhancedMonitoringInterval?: Duration;
 };
 
-export type DatabaseProps = MonitoringProps & {
+/**
+ * Database props without a VPC.
+ */
+export type ConfigurableDatabaseProps = MonitoringProps & {
+  /**
+   * The cluster identifier.
+   */
   clusterIdentifier: string;
+  /**
+   * The initial database name created.
+   */
   defaultDatabaseName: string;
+  /**
+   * Parameter group for the database.
+   */
   parameterGroupName: string;
+  /**
+   * Database username.
+   */
   username: string;
+  /**
+   * Database security group name.
+   */
+  securityGroupName: string;
+  /**
+   * Database secret name.
+   */
   masterSecretName: string;
+  /**
+   * Database engine version.
+   */
   version: rds.AuroraPostgresEngineVersion;
+  /**
+   * Number of database instances.
+   */
   numberOfInstance: number;
+  /**
+   * Min ACU for serverless database.
+   */
   minACU: number;
+  /**
+   * Max ACU for serverless database.
+   */
   maxACU: number;
+  /**
+   * Port to run the database on.
+   */
   dbPort: number;
+  /**
+   * The database removal policy.
+   */
   removalPolicy: RemovalPolicy;
-  allowedInboundSG?: ec2.SecurityGroup;
+  /**
+   * Create an inbound security group that can connect to the database. Stateless resources can recreate
+   * this security group to access the database.
+   */
+  inboundSecurityGroupName: string;
```

Comment:
Same concept; the inversion of control (IoC) here. 

Thinking of like; if you pass me along inbound SG (i.e. optional `allowedInboundSG?` ts trick) during db construction time then I will add this SG as source inbound allow rule to my private internal security.